rust/src/base.rs

```diff
@@ -539,6 +552,64 @@ impl FromStr for ConsignmentEndpoint {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, StrictEncode, StrictDecode)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+#[non_exhaustive]
+pub enum InvoiceChain {
+    /// Bitcoin mainnet
+    #[cfg_attr(feature = "serde", serde(rename = "mainnet"))]
```

Comment:
You do not need this if you add `#[serde(rename_all = "lowercase")]` at the structure level